### PR TITLE
Task: vv/74 add shownotes table

### PR DESCRIPTION
**Pull request by vitalik-prog**

### CODE REVIEW

**Review by ghost — CHANGES_REQUESTED** [review]
make shownotes as one word

**Review by EugeniusGR — DISMISSED** [review]

File: backend/src/api/episodes/episodes.api.ts [hunk 1]
```diff
@@ -27,7 +28,13 @@ const initEpisodesApi = ({ apiRouter, episodeService }: Args): Router => {
...
     EpisodesApiPath.$ID,
     handleAsyncApi(async (req, res) => {
-      return res.send(await episodeService.getById(req.params.id)).status(HttpCode.OK);
+      const episode = await episodeService.getById(req.params.id);
+      const timestamps = await shownoteService.getAllTimeNotesByEpisodeId(req.params.id);
+      const response = {
+        ...episode,
+        timestamps,
+      };
+      return res.send(response).status(HttpCode.OK);
```
Comment (ghost) [inline backend/src/api/episodes/episodes.api.ts]: remove back

File: backend/src/api/episodes/episodes.api.ts [hunk 2]
```diff
@@ -40,6 +47,14 @@ const initEpisodesApi = ({ apiRouter, episodeService }: Args): Router => {
     }),
   );
 
+  episodeRouter.post(
+    EpisodesApiPath.TIMESTAMP,
+    handleAsyncApi(async (req, res) => {
+      const timestamp = await shownoteService.create(req.body);
+      return res.json(timestamp).status(HttpCode.CREATED);
+    }),
+  );
```
Comment (ghost) [inline backend/src/api/episodes/episodes.api.ts]:
```suggestion
```

File: backend/src/common/enums/db/table-name.enum.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
 enum TableName {
   USERS = 'users',
   EPISODES = 'episodes',
+  SHOW_NOTES = 'show_notes',
```
Comment (ghost) [inline backend/src/common/enums/db/table-name.enum.ts]:
```suggestion
  SHOWNOTES = 'shownotes',
```

File: backend/src/data/repositories/shownote/shownote.repository.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+  constructor({ ShownoteModel }: Constructor) {
+    this.#ShownoteModel = ShownoteModel;
+  }
+
+  public create(payload: ShownoteCreatePayload): Promise<TShownote> {
+    return this.#ShownoteModel.query().insert(payload);
+  }
+
+  public getAllTimeNotesByEpisodeId(id: string): Promise<TShownote[]> {
```
Comment (ghost) [inline backend/src/data/repositories/shownote/shownote.repository.ts]:
```suggestion
  public getAllByEpisodeId(id: string): Promise<TShownote[]> {
```

File: backend/src/services/shownote/shownote.service.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+  constructor({ shownoteRepository }: Constructor) {
+    this.#shownoteRepository = shownoteRepository;
+  }
+
+  public create(payload: ShownoteCreatePayload): Promise<TShownote> {
+    return this.#shownoteRepository.create(payload);
+  }
+
+  public async getAllTimeNotesByEpisodeId(id: string): Promise<TShownote[]> {
```
Comment (ghost) [inline backend/src/services/shownote/shownote.service.ts]:
```suggestion
  public async getAllByEpisodeId(id: string): Promise<TShownote[]> {
```

File: shared/src/common/enums/api/episodes-api-path.enum.ts [hunk 6]
```diff
@@ -1,6 +1,7 @@
 enum EpisodesApiPath {
   ROOT = '/',
   $ID = '/:id',
+  TIMESTAMP = '/timestamp',
```
Comment (ghost) [inline shared/src/common/enums/api/episodes-api-path.enum.ts]:
```suggestion
```

**Review by ghost — CHANGES_REQUESTED** [review]

File: backend/src/api/episodes/episodes.api.ts [hunk 7]
```diff
@@ -8,8 +8,8 @@ import { handleAsyncApi } from '~/helpers/helpers';
 import { validateSchema as validateSchemaMiddleware } from '~/middlewares/middlewares';
 
 type Args = {
-  apiRouter: Router;
-  episodeService: typeof episodeService
+  apiRouter: Router,
+  episodeService: typeof episodeService,
```
Comment (ghost) [inline backend/src/api/episodes/episodes.api.ts]:
```suggestion
  apiRouter: Router;
  episodeService: typeof episodeService;
```

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**